File: Mario_Cash_pset1/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

float get_positive_float(string prompt);

int main(void)
{
    // Prompt user for float
    float dollars = get_positive_float("Change: ");
    
    // Convert dollars to cents
    int cents = round(dollars * 100);
    
    // Use the largest coin possible
    int q = 0;
    int d = 0;
    int n = 0;
    int p = 0;
    int count;
    
    // How many quarters can be used
    while (cents >= 25)
    {
        cents = cents - 25;
        q++;
    }
    
    // How many dimes can be used
    while (cents < 25 && cents >= 10)
    {
        cents = cents - 10;
        d++;
    }
    
    // How many nickels can be used
    while (cents < 10 && cents >= 5)
    {
        cents = cents - 5;
        n++;
    }
    
    // How many pennies can be used
    while (cents < 5 && cents >= 1)
    {
        cents = cents - 1;
        p++;
    }
    
    count = q + d + n + p;
    
    printf("%i\n", count);
}

// Prompt user for positive float
float get_positive_float(string prompt)
{
    float f;
    do
    {
        f = get_float("%s", prompt);
    }
    while (f <= 0);
    return f;
}

File: CS50/python_pset6/cash.py
from cs50 import get_float, get_int

# Prompt user for float
while True:
    dollars = get_float("Change: ")
    if dollars > 0:
        break

# Convert dollars to cents
cents = round(dollars * 100)

# Use the largest coin possible
q = 0
d = 0
n = 0
p = 0
count = 0

# How many quarters can be used
while (cents >= 25):
    cents = cents - 25
    q += 1

# How many dimes can be used
while (cents < 25 and cents >= 10):
    cents = cents - 10
    d += 1

# How many nickels can be used
while (cents < 10 and cents >= 5):
    cents = cents - 5
    n += 1

# How many pennies can be used
while (cents < 5 and cents >= 1):
    cents = cents - 1
    p += 1

count = q + d + n + p

print(count)